### ISSUE TITLE
Qwen3_Tech_Report

### ISSUE THREAD

**Issue by rainmaker712** [rainmaker712/ML-Research#8]
## 📌 제목

**Qwen3: Unified Reasoning & Response with Open-Weight Dense and MoE Models**

---

## 👨‍🔬 저자 및 팀

**Qwen Team**
Alibaba Group 산하 AI 연구 그룹, HuggingFace 및 ModelScope 통해 배포

---

## 🎯 연구 목적

* 다양한 추론 능력과 실시간 응답 요구를 동시에 만족시키는 LLM 개발
* Dense와 MoE 구조 모두 포함하며, 광범위한 파라미터 스케일(0.6B\~235B) 지원
* **Thinking / Non-thinking mode 통합** 및 **thinking budget** 개념 도입을 통해 유연한 추론 컨트롤 지원
* 강력한 성능을 유지하면서도 훈련 및 추론 비용을 절감

---

## 🧠 배경

* 최근 GPT-4o, Claude 3.7, Gemini 2.5 등 폐쇄형 모델들이 고성능을 보여주는 가운데,
  오픈소스 모델들도 빠르게 성능 격차를 줄이고 있음
* Qwen2.5의 후속으로서, 더 넓은 언어 범위와 정교한 reasoning capabilities 제공 목표

---

### 🔸 Dense 모델

| 모델명        | Layers | Heads (Q/KV) | Embedding 공유 | Context Length |
| ---------- | ------ | ------------ | ------------ | -------------- |
| Qwen3-0.6B | 28     | 16 / 8       | Yes          | 32K            |
| Qwen3-32B  | 64     | 64 / 8       | No           | 128K           |

### 🔸 MoE 모델

| 모델명             | Layers | Heads (Q/KV) | Experts (총 / 활성) | Context Length |
| --------------- | ------ | ------------ | ---------------- | -------------- |
| Qwen3-235B-A22B | 94     | 64 / 4       | 128 / 8          | 128K           |

* **공통 아키텍처 요소**: GQA, SwiGLU, RoPE, RMSNorm, QK-Norm
* **Tokenizer**: Byte-level BPE, vocab size 151,669

---

### 💾 데이터 구성

* 총 **36조 토큰**, **119개 언어 및 방언** 포함
* Qwen2.5-VL을 활용한 PDF 텍스트 추출
* 수학(Qwen2.5-Math), 코딩(Qwen2.5-Coder) 전용 모델로 합성 데이터 생성
* **세밀한 다차원 주석 시스템**을 활용해 데이터 품질 향상

### 🧩 3단계 사전 학습 전략

1. **General Stage** (30T tokens): 일반 언어 능력 학습
2. **Reasoning Stage** (5T tokens): STEM/코딩 비중 증가, LR decay 가속화
3. **Long Context Stage** (context 최대 32K): ABF, YARN, Dual Chunk Attention 등 도입

---

### 🔄 4단계 훈련 파이프라인

1. **Long-CoT Cold Start**
2. **Reasoning RL (GRPO)**
3. **Thinking Mode Fusion**: `/think` vs `/no think` 템플릿 통합
4. **General RL**: Instruction-following, Alignment, Agent 능력 등 강화

### 🧬 Distillation 전략

* **Strong-to-Weak Distillation**
* 대형 모델로부터 logits 기반 오프라인/온라인 distilled student 훈련
* 1/10의 GPU 시간으로 학습 효율 대폭 개선

---

### 🔹 표 1: Qwen3-235B-A22B vs 기존 모델

| 모델                  | MMLU     | GSM8K    | EvalPlus | INCLUDE  | AIME'24  |
| ------------------- | -------- | -------- | -------- | -------- | -------- |
| Qwen2.5-72B         | 86.1     | 91.5     | 65.9     | 69.1     | 74.3     |
| DeepSeek-V3         | 87.2     | 87.6     | 63.8     | 75.2     | 79.8     |
| **Qwen3-235B-A22B** | **87.8** | **94.4** | **77.6** | **73.5** | **85.7** |

### 🔹 표 2: Qwen3-32B vs Qwen2.5-72B / o3-mini

| 모델            | MMLU-Pro | SuperGPQA | EvalPlus | BFCL v3  | INCLUDE  |
| ------------- | -------- | --------- | -------- | -------- | -------- |
| Qwen2.5-72B   | 58.1     | 36.2      | 65.9     | 66.2     | 69.1     |
| o3-mini       | 56.1     | 34.5      | 70.6     | 62.9     | 75.1     |
| **Qwen3-32B** | **65.5** | **39.8**  | **72.1** | **72.5** | **67.9** |

---

## 📈 분석 및 결론

* **Thinking/Non-thinking 모드의 통합**은 하나의 모델로 두 역할을 수행하도록 하여 활용도를 크게 향상
* **Thinking Budget** 기능은 사용자 맞춤형 추론 비용 조절 가능
* **MoE 구조 개선**과 **효율적인 distillation 전략**을 통해 중소형 모델도 고성능 확보
* **Qwen3-235B-A22B**는 여러 reasoning benchmark에서 DeepSeek-R1, Grok-3-Beta, Gemini2.5-Pro와 유사한 수준의 성능

---

## ⚠️ 한계점

* Reasoning 능력은 여전히 토큰 budget 및 prompt 설계에 의존
* Non-thinking mode에서도 hallucination은 완전히 제거되지 않음
* Long-context reasoning task에 대한 성능 검증은 추가 실험이 필요

---

## 🔭 향후 연구 방향

* Token-efficient multi-modal 확장 (Vision, Audio 등)
* Domain-specific tuning을 위한 plug-in framework 구축
* More efficient MoE routing algorithm 도입
* Multi-turn agent simulation 향상

---

## 📎 Appendix 주요 내용 요약

* 다국어 평가 벤치마크: Multi-IF, INCLUDE, MMMLU, MT-AIME2024 등
* Evaluation 설정:

  * Thinking mode: temp=0.6, top-p=0.95
  * Non-thinking mode: temp=0.7, top-p=0.8
  * Max token: 기본 32K, AIME는 38,912까지 확장
* Chat template 예시:

  ```text
  /think → <think> ... </think>
  /no think → <think></think>
  ```

---
